packages/main/src/List.ts

```diff
@@ -641,7 +641,7 @@ class List extends UI5Element {
 	}
 
 	get hasData() {
-		return this.getItems().length !== 0;
+		return this.getItems().length !== 0 && !(this.getItems().length === 1 && this.getItems()[0].hasAttribute("ui5-drop-indicator"));
```

Comment:
The current DnD implementation seems incorrect to me, and the fix should align with how it is handled in the `ui5-tree`.  

For `ui5-list`, everything appears to function correctly and as expected.  

I suggest applying the fix to the `TreeList` component by enhancing the `getItems` method and addressing the implementation refactoring in a separate task.

Reply:
Follow up task: https://github.com/SAP/ui5-webcomponents/issues/11120